src/Symfony/Component/Process/Process.php

```diff
@@ -339,17 +347,9 @@ public function wait($callback = null)
      * Returns the Pid (process identifier), if applicable.
      *
      * @return int|null The process id if running, null otherwise
-     *
-     * @throws RuntimeException In case --enable-sigchild is activated
      */
     public function getPid()
     {
-        if ($this->isSigchildEnabled()) {
```

Comment:
Then it works :) Because the issue was the shell wrapper hack, and the new one is now able to report the correct pid. The exception should not have been thrown when in non-enhanced-sigchlid mode.